### DISCUSSION TITLE
Optuna and MySQL

### DISCUSSION THREAD

**Discussion by KevinCrp** [optuna/optuna#3920] [Q&A] [answered]
Hi I'm using Optuna and MySQL for distributed HPO but when I use the commands to create a study:

`mysql -u root -e "CREATE DATABASE IF NOT EXISTS example"` OK
`optuna create-study --study-name "distributed-example" --storage "mysql://root@localhost/example"`

With the second command i got the following error :
[E 2022-08-23 15:14:02,727] Failed to import DB access module for the specified storage URL. Please install appropriate one.

I'm using Ubuntu 18.04 on WSL2.

Any idea?

**Comment by himkt** [comment, marked as answer]
As the message suggested, please install a DB access module for the backend. If you want to use MySQL, you can use libraries such as [pymysql](https://github.com/PyMySQL/PyMySQL).

**Reply by KevinCrp** [reply]
This does not solve the problem

**Reply by nzw0301** [reply]
how about https://stackoverflow.com/a/66400594/2406562?

**Reply by KevinCrp** [reply]
I fix it with pymysql and by using `--storage "mysql+pymysql://root:root@localhost/example"` in the optuna create_study command.
But I got a new issue to access the database, I got permission denied whereas I can log to MySQL with that login and pwd.

**Reply by AHolliday** [reply]
I am having the same problem.  Installing mysqlclient does not help.  Can anyone assist?

**Reply by StackChan** [reply]
I encountered the same problem now. Any solutions after almost one year? Or we just use sqlite can help?

**Reply by StackChan** [reply]
I fixed the problem with the help of New Bing:
![image](https://github.com/optuna/optuna/assets/70527681/7f40084e-7874-4612-b74a-c1309009f5f4)
So login mysql,and run:
```
ALTER USER ‘root’@‘localhost’ IDENTIFIED WITH mysql_native_password BY ‘123456’;
```
then run
```
sudo service mysql start
```
Now I can implement parallelization easily with Optuna and Mysql!

**Comment by StackChan** [comment]
I fixed the problem with the help of New Bing:
![image](https://github.com/optuna/optuna/assets/70527681/2af67b1d-b3ae-4f47-a482-02383eff2e5e)
So login mysql,and run:
```
ALTER USER ‘root’@‘localhost’ IDENTIFIED WITH mysql_native_password BY ‘123456’;
```
then run
```
sudo service mysql start
```
Now I can implement parallelization easily with Optuna and Mysql!

Hope to be marked as the answer!
